### ISSUE TITLE
Merge "assumed-valid blocks" from Core

### ISSUE THREAD

**Issue by BlinkyStitt** [gandrewstone/bitcoin#324]
This change that is currently in the 0.14 RC looks like a great way to speed up initial block sync.

Any chances of merging it into BU?

https://www.reddit.com/r/Bitcoin/comments/5uy4h6/bitcoin_core_0140_release_candidate_1_available/ddy2f61/

https://github.com/bitcoin/bitcoin/blob/0.14/doc/release-notes.md#introduction-of-assumed-valid-blocks

**Comment by sickpig** [comment]
fwiw I find it a very clever way to decrease validation work during IBD/sync step because you can avoid checking sigs for ancestors to the assume-valid block. 
Basically is a new way to cut the validation work that it is immune to big reorg (contrary to what happens with checkpoints).

**Comment by kyuupichan** [comment]
Agreed.  If no one beats me to it I'll get round to this at some stage.

**Comment by ptschip** [comment]
why do we need this?

we already do this by not checking sigs unles blocks are less than 30
days old.

**Comment by kyuupichan** [comment]
There was a recent discussion (perhaps Slack; not sure) where I thought there was general agreement that the Core approach of making this a function of number of blocks, rather than time, was more robust.  I think @gandrewstone said he would accept a PR but wouldn't work on it himself.

Myself I think it is better, and would also like to minimize differences with the Core codebase, for as long as worthwhile.  Once we diverge sufficiently I'm happy to drop maintaining updates, but for now I think it benefits the project to minimize differences.

**Comment by ptschip** [comment]
i don't mind making our own a function of block height, but my
understanding of core's approach is that it's not auto updating , you
have to manually make the setting before starting IBD?  i may be wrong
that's what i remember...i think our approach being automated is better,
but we did have a discussion about changing from days to block height as
a cutoff...easy enough to do, i just don't see we need core code
here..our whole approach in BU is only 2 lines of code...it's not much
of a divergence to worry about IMO.

**Comment by nfujimoto** [comment]
I think the assumed-valid blocks is a better solution as stated for the reason above. Apart from the slight technical advantage, the concept is more align with BU ideology because it allows user to choose what they believe to be valid block. It is an option for user to choose the performance/security trade off.
